commit eb3b3264cd2c10720e92caf75671696859a88203
Author: Kilburn <kilburn@la3.org>
Date:   Sat Mar 2 11:02:49 2013 +0100

    Print rnspashot's PID when logging to syslog
    
    Print rsnapshot's PID instead of the logger's PID when logging to
    syslog. This is much more useful because it allow to relate log
    lines to rsnapshot instances when there are multiple concurrent
    ones. Previously each line had a different PID, even those
    generated by the same instance.
    
    The change is achieved by dropping the "-i" option from the
    logger's call and appending the rnapshot's PID to the tag itself.

diff --git a/rsnapshot-program.pl b/rsnapshot-program.pl
index 3d95cc2..9d9e50f 100755
--- a/rsnapshot-program.pl
+++ b/rsnapshot-program.pl
@@ -2098,15 +2098,15 @@ sub syslog_msg {
 	if (defined($config_vars{'cmd_logger'})) {
 		# print out our call to syslog
 		if (defined($verbose) && ($verbose >= 4)) {
-			print_cmd("$config_vars{'cmd_logger'} -i -p $facility.$level -t rsnapshot $msg");
+			print_cmd("$config_vars{'cmd_logger'} -p $facility.$level -t rsnapshot[$$] $msg");
 		}
 		
 		# log to syslog
 		if (0 == $test) {
-			$result = system($config_vars{'cmd_logger'}, '-i', '-p', "$facility.$level", '-t', 'rsnapshot', $msg);
+			$result = system($config_vars{'cmd_logger'}, '-p', "$facility.$level", '-t', "rsnapshot[$$]", $msg);
 			if (0 != $result) {
 				print_warn("Could not log to syslog:", 2);
-				print_warn("$config_vars{'cmd_logger'} -i -p $facility.$level -t rsnapshot $msg", 2);
+				print_warn("$config_vars{'cmd_logger'} -p $facility.$level -t rsnapshot[$$] $msg", 2);
 			}
 		}
 	}
